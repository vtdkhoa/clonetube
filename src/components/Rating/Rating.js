import React from 'react'
import { Icon, Progress } from 'semantic-ui-react'
import { formatNumber } from '../../services/number/number-format'
import PropTypes from 'prop-types'
import './Rating.scss'

const Rating = ({ likeCount, dislikeCount }) => {
  let ratingProgress = null
  let likeAmount = likeCount !== 0 ? likeCount : null
  let dislikeAmount = null

  if (likeCount && dislikeCount) {
    const amountLikes = parseFloat(likeCount)
    const amountDislikes = parseFloat(dislikeCount)
    const ratingPercent = 100.0 * (amountLikes / (amountLikes + amountDislikes))

    likeAmount = formatNumber(amountLikes)
    dislikeAmount = formatNumber(amountDislikes)
    ratingProgress = <Progress
      className="progress"
      percent={ratingPercent}
      size="tiny"
    />
  }

  return (
    <div className="rating">
      <div className="thumbs-up">
        <Icon name="thumbs up outline"/>
        <span>{likeAmount}</span>
      </div>
      <div className="thumbs-down">
        <Icon name="thumbs down outline"/>
        <span>{dislikeAmount}</span>
      </div>
      {ratingProgress}
    </div>
  )
}

Rating.propTypes = {
  likeCount: PropTypes.any,
  dislikeCount: PropTypes.any
}

export default Rating