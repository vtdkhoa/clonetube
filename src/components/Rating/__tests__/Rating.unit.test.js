import React from 'react'
import Rating from '../Rating'
import { shallow } from 'enzyme'

describe('Rating', () => {
  test('renders', () => {
    const wrapper = shallow(<Rating/>)
    expect(wrapper).toMatchSnapshot()
  })
})