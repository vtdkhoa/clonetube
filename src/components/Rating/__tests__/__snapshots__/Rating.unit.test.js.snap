// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs up outline"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs down outline"
    />
    <span />
  </div>
</div>
`;
